samples/net/sockets/big_http_download/src/big_http_download.c

```diff
@@ -99,30 +109,111 @@ ssize_t sendall(int sock, const void *buf, size_t len)
 	return 0;
 }
 
-int skip_headers(int sock)
+static int parse_status(bool *redirect)
+{
+	char *ptr;
+	int code;
+
+	ptr = strstr(response, "HTTP");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	ptr = strstr(response, " ");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	ptr++;
+
+	code = atoi(ptr);
+	if (code >= 300 && code < 400) {
+		*redirect = true;
+	}
+
+	return 0;
+}
+
+static int parse_header(bool *location_found)
+{
+	char *ptr;
+
+	ptr = strstr(response, "Location");
+	if (ptr == NULL) {
+		return 0;
+	}
+
+	ptr = strstr(response, " ");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	ptr++;
+
+	strncpy(download_url, ptr, sizeof(download_url));
+
+	ptr = strstr(download_url, "\r\n");
```

Comment:
It's been years since I made a http client but I do vaguely remember some servers not following this, e.g., using `location:` with a lower-case l, or using \n line endings only - does this sample need to cater for those?

Reply:
It shouldn't be too difficult to handle those cases, I'll rewrite the parser to cover that.

Reply:
Both mentioned issues should be covered now.